src/category_theory/monoidal/discrete.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.monoidal.natural_transformation
+import category_theory.discrete_category
+import algebra.group.hom
+
+/-!
+# Monoids as discrete monoidal categories
+
+The discrete category on a monoid is a monoidal category.
+Multiplicative morphisms induced monoidal functors.
+-/
+
+universes u
+
+open category_theory
+
+variables (M : Type u) [monoid M]
+
+namespace category_theory
+
+instance monoid_discrete : monoid (discrete M) := by { dsimp [discrete], apply_instance }
+
+instance : monoidal_category (discrete M) :=
+{ tensor_unit := 1,
+  tensor_obj := λ X Y, X * Y,
+  tensor_hom := λ W X Y Z f g, ⟨⟨by rw [f.1.1, g.1.1]⟩⟩,
```

Comment:
Since there's a bunch of `.1.1` deconstructors and `⟨⟨_⟩⟩` constructors, perhaps it's a good idea to have named versions for these

Reply:
Good point! I added `eq_of_hom`, and use `eq_to_hom` instead of `⟨⟨_⟩⟩`. I also took the opportunity to use `hom_of_le` in a few places.